fenix/app/src/main/java/org/mozilla/fenix/compose/LinkText.kt

```diff
@@ -0,0 +1,108 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import mozilla.components.support.ktx.android.content.isScreenReaderEnabled
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * The tag used for links in the text for annotated strings.
+ */
+private const val URL_TAG = "URL_TAG"
+
+/**
+ * Model containing link text, url and action.
+ */
+data class LinkTextState(
+    val text: String,
+    val url: String,
+    val onClick: (String) -> Unit,
+)
+
+/**
+ * A composable for displaying text that contains a clickable link text.
+ *
+ * @param text The complete text.
+ * @param linkTextState The clickable part of the text.
+ */
+@Composable
+fun LinkText(
+    text: String,
+    linkTextState: LinkTextState,
+) {
+    val context = LocalContext.current
+    val annotatedString = buildAnnotatedString {
+        val startIndex = text.indexOf(linkTextState.text, ignoreCase = true)
+        val endIndex = startIndex + linkTextState.text.length
+        append(text)
+        addStyle(
+            style = SpanStyle(color = FirefoxTheme.colors.textAccent),
+            start = startIndex,
+            end = endIndex,
+        )
```

Comment:
```suggestion

        append(text)

        addStyle(
            style = SpanStyle(color = FirefoxTheme.colors.textAccent),
            start = startIndex,
            end = endIndex,
        )
```
I know this is just a simple refactoring here. Could we add some new lines here to help make this more readable while we are at it? 🙇